akka-http-core/src/main/scala/akka/http/scaladsl/ClientTransportWithCustomResolver.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import java.net.InetSocketAddress
+
+import akka.actor.ActorSystem
+import akka.annotation.ApiMayChange
+import akka.http.scaladsl.Http.OutgoingConnection
+import akka.http.scaladsl.settings.ClientConnectionSettings
+import akka.stream.scaladsl.{ Flow, Tcp }
+import akka.util.ByteString
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+@ApiMayChange
+trait ClientTransportWithCustomResolver extends ClientTransport {
```

Comment:
Should we make it a class/trait based API? Otherwise, it could be just a

```scala
def withCustomResolver(resolve: (String, Int) => Future[InetSocketAddress]): ClientTransport
```

or create an extra `trait` just for the resolution part if we want to give it a name and more parameters.